src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,74 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exeption.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+        user.setLogin("Daedrus");
+        user.setAge(25);
+        user.setPassword("Daedrus123");
+
+    }
+
+    @Test
+    void nullUser_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void nullLogin_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void nullPassword_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void nullAge_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void user_Age_NotOk() {
+        user.setAge(17);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void user_shortLogin_NotOk() {
+        user.setLogin("Jonn");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
```

Comment:
The 'user_shortLogin_NotOk' test seems to be checking for a short login, but there is no specified minimum login length in the checklist. Ensure that the test case aligns with the task requirements.